### ISSUE TITLE
Fine-tuning BEATs for ESC50 classification.

### ISSUE THREAD

**Issue by e871223eeee** [microsoft/unilm#1112]
I'm trying to replicate your experiment on fine-tuning for ESC50 classification, but I'm experiencing a significant accuracy gap. I'm using the pre-trained model "BEATs_iter3" and I can't achieve an accuracy of 95.6%. In fact, it's not even reaching 90%. I would like to request your code for this task or any specific details that I should pay special attention to. Thank you for your assistance, and I appreciate your prompt response.
![螢幕擷取畫面 2023-06-03 203007](https://github.com/microsoft/unilm/assets/58674364/e9c11918-ffd6-4726-918e-853517e66c67)

**Comment by Derdin-datascience** [comment]
Having same problem here

**Comment by jmren168** [comment]
Any ideas about how to re-train (or fine-tune) a pre-trained model for a specific dataset? Many thanks.

**Comment by Adamwewe** [comment]
Hello @e871223eeee! Have you made sure to use `torch.no_grad()` to freeze layers preceding the final layer? If not you could be accidentally retraining on ESC-50 instead of fine tuning to it which could potentially explain the mismatch in accuracy.

**Comment by jromerooo2** [comment]
@e871223eeee https://github.com/NINAnor/rare_species_detections/tree/main/BEATs_on_ECS50 maybe that could help

**Comment by BenCretois** [comment]
@jromerooo2 thanks for sharing our repo! However the link is broken and the correct one is now: https://github.com/NINAnor/rare_species_detections/tree/main/BEATs_on_ESC50

A few things:

- We recently made some changes in the [repo](https://github.com/NINAnor/rare_species_detections/tree/main/BEATs_on_ESC50)
- The repo is fully Dockerized and self-contained (you don't need to download BEATs or the ESC50 dataset) so you should be able to run it without problems (I tested it on different Linux distribution)
- We find that training only the last fully connected layer yield better results than training the full architecture (last FC layer + BEATs)

If you have any problem running our code do not hesitate to open an issue :)

**Comment by XiaokangY** [comment]
I would like to ask if I want to further input the extracted features into the resnet18 pre-trained model for further feature mining, I found that the loss cannot be reduced in this way. Please help me.

**Comment by Shikhar-S** [comment]
> [@jromerooo2](https://github.com/jromerooo2) thanks for sharing our repo! However the link is broken and the correct one is now: https://github.com/NINAnor/rare_species_detections/tree/main/BEATs_on_ESC50
> 
> A few things:
> 
> * We recently made some changes in the [repo](https://github.com/NINAnor/rare_species_detections/tree/main/BEATs_on_ESC50)
> * The repo is fully Dockerized and self-contained (you don't need to download BEATs or the ESC50 dataset) so you should be able to run it without problems (I tested it on different Linux distribution)
> * We find that training only the last fully connected layer yield better results than training the full architecture (last FC layer + BEATs)
> 
> If you have any problem running our code do not hesitate to open an issue :)

Hi, could you please lmk what is the final accuracy you get on ESC50 with your training pipeline?

**Comment by andrew-aladjev** [comment]
> https://github.com/NINAnor/rare_species_detections/tree/main/BEATs_on_ESC50

You are missing very important thing: `extract_features` method has `fbank_mean` and `fbank_std`. You need to calculate those values based on your dataset. You can search for `AST` model and find out how to implement it.
